src/lib/bootloader/stage1_proposal.rb

```diff
@@ -146,24 +152,30 @@ def boot_with_btrfs?
 
       def init_boot_info
         return if @boot_initialized
-
         @boot_initialized = true
-        boot_disk_map = Yast::Storage.GetTargetMap[Yast::BootStorage.disk_with_boot_partition] || {}
-        boot_part = Yast::Storage.GetPartition(Yast::Storage.GetTargetMap,
-          Yast::BootStorage.BootPartitionDevice)
-        @logical_boot = boot_part["type"] == :logical
-        @boot_with_btrfs = boot_part["used_fs"] == :btrfs
+
+        boot_part = devicegraph.partitions.with(name: Yast::BootStorage.BootPartitionDevice).first
+        @logical_boot = boot_part.type == Storage::PartitionType_LOGICAL
+        @boot_with_btrfs = with_btrfs?(boot_part)
 
         # check for sure also underlaying partitions
-        (boot_disk_map["partitions"] || []).each do |p|
-          @extended = p["device"] if p["type"] == :extended
-          next unless underlaying_boot_partition_devices.include?(p["device"])
+        disk_name = Yast::BootStorage.disk_with_boot_partition
+        devicegraph.disks.with(name: disk_name).partitions.each do |p|
+          @extended = p.name if p.type == Storage::PartitionType_EXTENDED
+          next unless underlaying_boot_partition_devices.include?(p.name)
 
-          @boot_with_btrfs = true if p["used_fs"] == :btrfs
-          @logical_boot = true if p["type"] == :logical
+          @boot_with_btrfs = with_btrfs?(p)
+          @logical_boot = true if p.type == Storage::PartitionType_LOGICAL
         end
       end
 
+      def with_btrfs?(partition)
+        partition.filesystem.type == ::Storage::FsType_BTRFS
+      rescue Storage::WrongNumberOfChildren
+        # No filesystem in the partition
+        false
+      end
```

Comment:
hmmm, previously it looks better for me, to be honest I still do not like new API for it.